### PR TITLE
Add description text functionality to core-radio & shared-choice

### PR DESCRIPTION
**Pull request by marcod1419**
See #746

This PR adds description functionality to core-radio. It allows a string to be passed into a new `description` prop that will generate a line of descriptive text below the radio input.

### CODE REVIEW

File: packages/Radio/Radio.md [hunk 1]
```diff
@@ -43,3 +44,41 @@ const setChoice = event => {
   />
 </Box>
 ```
+
+### Radio descriptions
+
+In special circumstances, it may be impossible for the radio label to be descriptive enough for the user to make an informed choice. In these cases, use the `description` prop. This prop adds a string of text under a particular radio input that further describes its choice.
```
Comment (theetrain) [inline packages/Radio/Radio.md]: > This prop `can be used to add descriptive text` under a particular radio input...

File: shared/components/Choice/Choice.modules.scss [hunk 2]
```diff
@@ -78,3 +78,8 @@
   position: absolute;
   top: auto;
 }
+
+.descriptionText {
+  margin-top: 0.5rem;
```
Comment (theetrain) [inline shared/components/Choice/Choice.modules.scss]: Should this be managed by Box?

File: shared/components/Choice/Choice.jsx [hunk 3]
```diff
@@ -49,8 +49,13 @@ const renderColoredLabel = (color, content) => (
   </ColoredTextProvider>
 )
 
-const renderLabel = (label, feedback, checked, disabled) => {
-  const content = <Text size="medium">{label}</Text>
+const renderLabel = (label, description, feedback, checked, disabled) => {
```
Comment (theetrain) [inline shared/components/Choice/Choice.jsx]: Adding the `description` prop would technically make this available for core-checkbox. Should we add a separate story to work on adding description text for core-checkbox?

Comment (marcod1419) [reply shared/components/Choice/Choice.jsx]: I think there's potential for that, but we should consult design.

Comment (jraff) [reply shared/components/Choice/Choice.jsx]: Good catch, I think it's OOS for this fix. We can always add this as a feature to @tds/core-checkbox in the future.

**Pull request merged**